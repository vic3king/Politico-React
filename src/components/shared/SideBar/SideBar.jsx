import React from 'react';
import PropTypes from 'prop-types';
import '../../../style/sidebar.scss';
import ProfileImage from '../Profile/ProfileImage';
import ProfileDetails from '../Profile/ProfileDetails';

const SideBar = ({ ButtonOne, ButtonTwo, ButtonThree, ButtonFour, user }) => {
  return (
    <React.Fragment>
      <div className="profile sidenav toggle">
        <ProfileImage className="profile__image" />
        <ProfileDetails className="profile__details" user={user} />
        <br />
        <br />
        {ButtonOne}
        <br />
        <br />
        <br />
        {ButtonTwo}
        <br />
        <br />
        <br />
        {ButtonFour}
        <br />
        <br />
        <br />
        {ButtonThree}
      </div>
    </React.Fragment>
  );
};

SideBar.defaultProps = {
  ButtonOne: (PropTypes.defaultProps = ''),
  ButtonTwo: (PropTypes.defaultProps = ''),
  ButtonThree: (PropTypes.defaultProps = ''),
  ButtonFour: (PropTypes.defaultProps = ''),
};

SideBar.propTypes = {
  ButtonOne: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  ButtonTwo: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  ButtonThree: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
  ButtonFour: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
};

export default SideBar;
