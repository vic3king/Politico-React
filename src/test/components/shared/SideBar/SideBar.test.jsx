import React from 'react';
import { shallow } from 'enzyme';
import SideBar from '../../../../components/shared/SideBar/SideBar';

describe('SideBar component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a div tag', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper.find('div'));
  });
});
