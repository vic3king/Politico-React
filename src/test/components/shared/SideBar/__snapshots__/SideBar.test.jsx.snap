// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar
    ButtonFour=""
    ButtonOne=""
    ButtonThree=""
    ButtonTwo=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="profile sidenav toggle"
      >
        <ProfileImage
          className="profile__image"
        />
        <ProfileDetails
          className="profile__details"
        />
        <br />
        <br />
        
        <br />
        <br />
        <br />
        
        <br />
        <br />
        <br />
        
        <br />
        <br />
        <br />
        
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProfileImage
            className="profile__image"
          />,
          <ProfileDetails
            className="profile__details"
          />,
          <br />,
          <br />,
          "",
          <br />,
          <br />,
          <br />,
          "",
          <br />,
          <br />,
          <br />,
          "",
          <br />,
          <br />,
          <br />,
          "",
        ],
        "className": "profile sidenav toggle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "profile__image",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "profile__details",
            "user": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "",
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="profile sidenav toggle"
        >
          <ProfileImage
            className="profile__image"
          />
          <ProfileDetails
            className="profile__details"
          />
          <br />
          <br />
          
          <br />
          <br />
          <br />
          
          <br />
          <br />
          <br />
          
          <br />
          <br />
          <br />
          
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ProfileImage
              className="profile__image"
            />,
            <ProfileDetails
              className="profile__details"
            />,
            <br />,
            <br />,
            "",
            <br />,
            <br />,
            <br />,
            "",
            <br />,
            <br />,
            <br />,
            "",
            <br />,
            <br />,
            <br />,
            "",
          ],
          "className": "profile sidenav toggle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "profile__image",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "profile__details",
              "user": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "",
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
